spec/requests/topics_controller_spec.rb

```diff
@@ -1792,6 +1792,22 @@ def fabricate_topic(user, category = nil)
     end
   end
 
+  describe '#show_by_external_id' do
+    before_all do
+      Fabricate(:topic, external_id: 'asdf')
+    end
+
+    it 'returns 301 when found' do
+      get "/t/external_id/asdf.json"
+      expect(response.status).to eq(301)
```

Comment:
I don't think asserting for the response is sufficient here since we need to know whether the redirect was correct as well.